### PR TITLE
feat: add `--gateway-discovery-dns-strategy` flag

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

This PR introduces `--gateway-discovery-dns-strategy` flag which allow users to specify which strategy to use when creating address of Admin API endpoints.

There are 3 options that are accepted as values for this flag:

- `service` - for service scoped pod dns names: `pod-ip-address.service-name.my-namespace.svc.cluster-domain.example`
- `pod` - for namespace scope pod dns names: `pod-ip-address.my-namespace.pod.cluster-domain.example`
- `ip` (default, retains v2.9 behavior) - for regular IP addresses 

**Which issue this PR fixes**:

closes: #4065

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: config/variants/multi-gw/base/manager_multi_gateway_patch.yaml [hunk 1]
```diff
@@ -13,5 +13,7 @@ spec:
         env:
         - name: CONTROLLER_KONG_ADMIN_SVC
           value: kong/kong-admin
+        - name: CONTROLLER_GATEWAY_DISCOVERY_DNS_STRATEGY
+          value: ip
```
Comment (pmalek) [inline config/variants/multi-gw/base/manager_multi_gateway_patch.yaml:17]: Do we want this explicitly in the manifest or do we prefer to rely on default defined in code? (and documented in flag's description)?

Comment (pmalek) [reply config/variants/multi-gw/base/manager_multi_gateway_patch.yaml]: Fair enough. Removed!

**Review by czeslavo — APPROVED** [review]

**Review by czeslavo — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -125,13 +125,25 @@ Adding a new version? You'll need three changes:
...
   `KongConfigurationApplyFailed` reason is recorded.
   [#4054](https://github.com/Kong/kubernetes-ingress-controller/pull/4054)
 - Disable translation to expression routes when feature gate `ExpressionRoutes`
   is enabled but feature gate `CombinedRoutes` is not enabled.
   [#4057](https://github.com/Kong/kubernetes-ingress-controller/pull/4057)
+- Added `--gateway-discovery-dns-strategy` flag which allows specifying which
+  DNS strategy to use when generating Gateway's Admin API addresses.
+  [#4071](https://github.com/Kong/kubernetes-ingress-controller/pull/4071)
+
+  There are 3 options available
```
[the remark this answers is not in the document]
Comment (pmalek) [reply CHANGELOG.md:139]: What do you mean here @randmonkey 🤔 ? The empty line and the indentation ensures that we have this entry in the proper place (please check the preview).

Feel free to suggest a change that you'd feel be better and we can discuss it.

[the remark this answers is not in the document]
Comment (randmonkey) [reply CHANGELOG.md:139]: OK, the preview seems good.

**Review by randmonkey — APPROVED** [review]

**Pull request merged**